src/juju_doctor/probes.py

```diff
@@ -0,0 +1,53 @@
+"""Helper module to wrap and execute probes."""
+
+import importlib.util
+import inspect
+import logging
+from dataclasses import dataclass
+from pathlib import Path
+
+from rich.logging import RichHandler
+
+from juju_doctor.artifacts import Artifacts
+
+SUPPORTED_PROBE_TYPES = ["status", "bundle", "show_unit"]
+
+logging.basicConfig(level=logging.WARN, handlers=[RichHandler()])
+log = logging.getLogger(__name__)
+
+
+@dataclass
+class Probe:
+    """A probe that can be executed via juju-doctor."""
+
+    name: str
+    uri: str
+    original_path: Path  # path in the source folder
+    path: Path  # path in the temporary folder
+
+    def run(self, artifacts: Artifacts):
+        """Run the probe."""
```

Comment:
A better docstring would be appreciated here:
```suggestion
        """Dynamically load a Python script from self.path and execute it, making its functions available.
        
        Execute these functions if they match: `status`, `bundle`, or `show_unit`.
        """
```
Loops through the functions, .